packages/option/dev/index.tsx

```diff
@@ -0,0 +1,56 @@
+import { Component, createSignal, For, Show } from "solid-js";
+import { createStore } from "solid-js/store";
+import { render } from "solid-js/web";
+import "uno.css";
+import { createOption, makeOption, None, Some } from "../src";
+
+const UserExample: Component = () => {
+  const [user, setUser] = createStore({
+    role: Some(1),
+    submissionValue: None<string>()
+  });
+  return (
+    <>
+      <p>Wrote: {user.submissionValue.unwrap_or("Didnt write yet.")}</p>
+      <input
+        type="text"
+        onInput={e => setUser("submissionValue", makeOption((e.target as any).value || null))}
```

Comment:
In solid `currentTarget` has the current ref assigned to the event so it also has the correct types.
```suggestion
        onInput={e => setUser("submissionValue", makeOption(e.currentTarget.value || null))}
```